app/dashboard/views.py

```diff
@@ -968,10 +968,27 @@ def scenario_visualize_cpn_stacked_timeseries(request, scen_id):
             )
         )
 
+    color_mapping = {
+        "pv_plant_flow": "#F2CD5D",
+        "battery_charge_flow": "#12AB6D",
+        "battery_discharge_flow": "#71D0A1",
+        "total_demand_flow": "#A69F99",
+        "fulfilled_demand_flow": "#716A64",
+        "diesel_generator_flow": "#814400",
+        "excess_flow": "#EA9822",
+        "ac_bus_excess_flow": "#EA9822",
+        "dc_bus_excess_flow": "#EA9822",
+    }
+
+    timeseries_labels = [f"{ts['label']}_flow" for ts in results_json[1]["data"][0]["timeseries"]]
```

Comment:
Why only in `results_json[1]` (in my case only gas) ? @paulapreuss

Reply:
Ah, for me 0 was gas and 1 was Electricity. Can this be random? To be honest I don't 100% see through the open plan functions used to construct the jsons for the graph